src/OrchardCore/OrchardCore/Extensions/ExtensionManager.cs

```diff
@@ -152,7 +147,111 @@ public IEnumerable<IFeatureInfo> GetDependentFeatures(string featureId)
+
+                var applicationContext = _serviceProvider.GetRequiredService<IApplicationContext>();
+                var typeFeatureProvider = _serviceProvider.GetRequiredService<ITypeFeatureProvider>();
+                var featuresProvider = _serviceProvider.GetRequiredService<IFeaturesProvider>();
+
+                var extensionDependencyStrategies = _serviceProvider.GetServices<IExtensionDependencyStrategy>();
+                var extensionPriorityStrategies = _serviceProvider.GetServices<IExtensionPriorityStrategy>();
+
+                var modules = applicationContext.Application.Modules;
+                var loadedExtensions = new ConcurrentDictionary<string, ExtensionEntry>();
+
+                // Load all extensions in parallel
+                Parallel.ForEach(modules, (module, cancellationToken) =>
+                {
+                    if (!module.ModuleInfo.Exists)
+                    {
+                        return;
+                    }
+
+                    var manifestInfo = new ManifestInfo(module.ModuleInfo);
+                    var extensionInfo = new ExtensionInfo(module.SubPath, manifestInfo, (mi, ei) =>
+                    {
+                        return featuresProvider.GetFeatures(ei, mi);
+                    });
+
+                    var entry = new ExtensionEntry
+                    {
+                        ExtensionInfo = extensionInfo,
+                        Assembly = module.Assembly,
+                        ExportedTypes = module.Assembly.ExportedTypes
+                    };
+
+                    loadedExtensions.TryAdd(module.Name, entry);
+                });
+
+                // Get all types from all extension and add them to the type feature provider.
+                foreach (var loadedExtension in loadedExtensions)
+                {
+                    var extension = loadedExtension.Value;
+
+                    foreach (var exportedType in extension.ExportedTypes.Where(IsComponentType))
+                    {
+                        if (!SkipExtensionFeatureRegistration(exportedType))
+                        {
+                            var sourceFeature = GetSourceFeatureNameForType(exportedType, extension.ExtensionInfo.Id);
+
+                            var feature = extension.ExtensionInfo.Features.FirstOrDefault(f => f.Id == sourceFeature);
+
+                            if (feature == null)
+                            {
+                                // Type has no specific feature, add it to all features
+                                foreach (var curFeature in extension.ExtensionInfo.Features)
+                                {
+                                    typeFeatureProvider.TryAdd(exportedType, curFeature);
+                                }
+                            }
+                            else
+                            {
+                                typeFeatureProvider.TryAdd(exportedType, feature);
+                            }
```

Comment:
maybe drop the `else` here using something like this

```suggestion
                            if (feature != null)
                            {
                                typeFeatureProvider.TryAdd(exportedType, feature);
                            
                                continue;
                            }
                            
                            // Type has no specific feature, add it to all features.
                            foreach (var curFeature in extension.ExtensionInfo.Features)
                            {
                                typeFeatureProvider.TryAdd(exportedType, curFeature);
                            }
```

Reply:
For me this makes it harder to read.

Reply:
okay. It's a preference. For me when I see multiple `else`, it makes the code harder to read. But, I understand if you like using else :)